File: ios/Anemone SDK/service/ITenantService.swift
//
//  ITenantService.swift
//  Anemone SDK
//
//  Created by Angel Luis Garcia on 14/02/2017.
//  Copyright © 2017 Syltek Solutions S.L. All rights reserved.
//

import Foundation

public protocol ITenantService {
    func search(name: String, pagination: PaginationOptions?) -> Promise<[Tenant]>
    func search(coordinate: Coordinate, radius: Int, sportId: SportId?, pagination: PaginationOptions?) -> Promise<[Tenant]>
    func search(tenantIds: [TenantId]) -> Promise<[Tenant]>
    func search(googlePlaceId: String) -> Promise<[Tenant]>

    func fetchDetail(id: TenantId) -> Promise<Tenant>
}

File: android/app/src/main/java/com/playtomic/test/anemone_sdk/service/ITenantService.kt
package com.anemonesdk.service

import com.anemonesdk.model.config.SportId
import com.anemonesdk.model.tenant.Tenant
import com.anemonesdk.model.tenant.TenantId
import com.anemonesdk.service.request.PaginationOptions
import com.playtomic.foundation.model.Coordinate
import com.playtomic.foundation.promise.Promise

/**
 * Created by agarcia on 15/02/2017.
 */

interface ITenantService {

    fun search(name: String, pagination: PaginationOptions?): Promise<List<Tenant>>
    fun search(coordinate: Coordinate, radius: Int, sportId: SportId?, pagination: PaginationOptions?): Promise<List<Tenant>>
    fun search(tenantIds: List<TenantId>): Promise<List<Tenant>>
    fun search(googlePlaceId: String): Promise<List<Tenant>>

    fun fetchDetail(id: TenantId): Promise<Tenant>
}
